hollow/src/main/java/com/netflix/hollow/api/consumer/fs/HollowFilesystemBlobRetriever.java

```diff
@@ -120,6 +144,7 @@ private void ensurePathExists(Path blobStorePath) {
 
     @Override
     public HollowConsumer.HeaderBlob retrieveHeaderBlob(long desiredVersion) {
+        cleanupOldBlobs();
```

Comment:
We could scope to deletion to a prefix for e.g. "header-" here, so we're a bit closer to only deleting files we created